src/main/java/org/prebid/server/validation/RequestValidator.java

```diff
@@ -163,78 +160,51 @@ private static String format(BigDecimal value) {
         }
 
-        for (final ExtPriceGranularityBucket bucket : buckets) {
-            validateGranularityBucket(bucket);
+        final Integer precision = extPriceGranularity.getPrecision();
+        if (precision != null && precision < 0) {
+            throw new ValidationException("Price granularity error: precision must be non-negative");
         }
-        validateGranularityBuckets(buckets);
+
+        validateGranularityRanges(ranges);
     }
 
     /**
-     * Validates {@link ExtPriceGranularityBucket}.
+     * Validates {@link ExtGranularityRange}s increment
      */
-    private static void validateGranularityBucket(ExtPriceGranularityBucket bucket) throws ValidationException {
-        final BigDecimal min = bucket.getMin();
-
-        if (min != null && bucket.getMax().compareTo(min) < 0) {
-            throw new ValidationException("Price granularity error: max must be greater than min");
-        }
-
-        if (bucket.getIncrement().compareTo(BigDecimal.ZERO) <= 0) {
+    private static void validateGranularityRangeIncrement(ExtGranularityRange range) throws ValidationException {
+        if (range.getIncrement().compareTo(BigDecimal.ZERO) <= 0) {
             throw new ValidationException("Price granularity error: increment must be a nonzero positive number");
         }
-
-        final Integer precision = bucket.getPrecision();
-        if (precision != null && precision < 0) {
-            throw new ValidationException("Price granularity error: precision must be non-negative");
-        }
-
     }
 
     /**
      * Validates {@link List<ExtRequestTargeting>} as set of ranges.
      */
-    private static void validateGranularityBuckets(List<ExtPriceGranularityBucket> buckets) throws ValidationException {
-        final Iterator<ExtPriceGranularityBucket> bucketIterator = buckets.iterator();
-        ExtPriceGranularityBucket bucket = bucketIterator.next();
-        final Integer precision = bucket.getPrecision();
-
-        while (bucketIterator.hasNext()) {
-            final ExtPriceGranularityBucket nextPriceGranularityBucket = bucketIterator.next();
-            if (!Objects.equals(precision, nextPriceGranularityBucket.getPrecision())) {
-                throw new ValidationException("Price granularity error: precision not consistent across entries");
-            }
-
-            if (bucket.getMax().compareTo(nextPriceGranularityBucket.getMax()) > 0) {
+    private static void validateGranularityRanges(List<ExtGranularityRange> ranges) throws ValidationException {
+        final Iterator<ExtGranularityRange> rangeIterator = ranges.iterator();
+        ExtGranularityRange range = rangeIterator.next();
+        validateGranularityRangeIncrement(range);
+        while (rangeIterator.hasNext()) {
+            final ExtGranularityRange nextGranularityRange = rangeIterator.next();
+            if (range.getMax().compareTo(nextGranularityRange.getMax()) > 0) {
```

Comment:
I believe there might be a bug here: this check always compares with first range's max since `range` is not updated to point to `nextGranularityRange` anywhere.